### PR TITLE
Feature/unit testing page

### PR DESCRIPTION
**Pull request by Deco354**
Created Unit Testing blog post and styled it.
Code snippets are from 3rd party library [Splash](https://github.com/JohnSundell/Splash)

### CODE REVIEW

File: unit-testing-best-practices.html [hunk 1]
```diff
@@ -0,0 +1,179 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
```
Comment (skirsimae) [inline unit-testing-best-practices.html:6]: Also add other meta tags such as description and keywords. :) More info: https://www.wordstream.com/meta-tags

File: unit-testing-best-practices.html [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+  <header class="home-banner">
+    <h1 class="blog-title">Swifty</h1>
+    <h2 class="blog-description">Weekly swift articles covering actual useful skills with short concise posts</h2>
+  </header>
+
+  <article class="container blog-article">
+    <header>
+      <div class="date-stamp">
+        <time datetime="2019-06-29">Jun 29th 2019</time> -
+        <address class="author"><a rel="author" href="">Declan McKenna</a></address>
```
Comment (skirsimae) [inline unit-testing-best-practices.html:24]: Perhaps you will do it later but the href is empty, add an url?

File: unit-testing-best-practices.html [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+      </div>
+      <h1>Unit Testing <br class="responsive-break">
```
Comment (skirsimae) [inline unit-testing-best-practices.html]: `responsive-break` class does not seems to have any css assigned?

File: unit-testing-best-practices.html [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+    <h2>Smaller Tests</h2>
+    <p>Minimize the number of asserts and test just one concept per test function. Smaller tests are easier to read and
+      give clearer failure results. Any repetitive setup involved in doing this can be handled by <code>setup()</code> or
+      private helper functions</p>
+
+    <figure>
+      <pre>
+      <code>
+<span class="keyword">func</span> testAddMonths() {
```
Comment (skirsimae) [inline unit-testing-best-practices.html:116]: Indentation a bit off.

**Review by skirsimae — COMMENTED** [review]
The website looks good and is responsive 👍 
The code looks also good, not much else to add other than some small comments such as add some more meta tags so that your blog would pop up in Google search :) I would perhaps also divide the css files per screen `index.css` `unit-testing-best-practies.css` and have common file(s) for stuff you reuse like for example you have for the `swift-code.css` (I can imagine that you will reuse it for other posts). Otherwise the the `styles.css` can become really big.

File: unit-testing-best-practices.html (hunk 4 above)
Comment (Deco354) [reply unit-testing-best-practices.html:116]: The weird thing is when I try and correct this it actually alters the code. I'm using a third party tool to generate the snippet HTML although it appears anything inside `<pre>` the indentation actually impacts the appearance of the rendered html 😨 

Bit annoying as it looks pretty bad.

File: unit-testing-best-practices.html (hunk 1 above)
Comment (Deco354) [reply unit-testing-best-practices.html:6]: Good idea, I'll do a separate PR and do this for all my existing pages. 
I'll dedicate a PR to the css files too the styles.css is getting pretty messy 😅

File: unit-testing-best-practices.html (hunk 4 above)
Comment (skirsimae) [reply unit-testing-best-practices.html:116]: Ah indeed, it does affect the layout.

**Comment by sandoche** [comment]
I didn't check the page itself I am waiting for you to put it to netlify and share the link.

A few recommendations:
- You may want to use a CSS naming convention, I personally use BEM, also the most used: https://2019.stateofcss.com/technologies/methodologies/
- Always check (to be honest I don't always do, only when I want to optimize the SEO) with Google Lighthouse (it's in the audit tab of the developer's console)
- Check out this (before launching a website) : http://weblaunchchecklist.com/

**Comment by sandoche** [comment]
> The website looks good and is responsive 👍
> The code looks also good, not much else to add other than some small comments such as add some more meta tags so that your blog would pop up in Google search :) I would perhaps also divide the css files per screen `index.css` `unit-testing-best-practies.css` and have common file(s) for stuff you reuse like for example you have for the `swift-code.css` (I can imagine that you will reuse it for other posts). Otherwise the the `styles.css` can become really big.

Actually when you deploy your site it's better to have one big file because it's loaded only once.
And some people use the Critical Path to put a bit of CSS on the HTML directly to optimize it. 

But when you develop to practice it's better to separate in a lot of files. Because when you do frontend for real you use most of the time a CSS Processors (like Sass) and you split your code in many files and then at the end it compiles it. If your file is really to big frontend people uses webpack that will cut it in chunks per page automatically so you don't have to take care of it.

**Pull request merged**